src/components/LoginModal/LoginModal.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect } from "react";
+import ModalWithForm from "../ModalWithForm/ModalWithForm";
+
+const LoginModal = ({
+  handleCloseModal,
+  onLogin,
+  isOpen,
+  onSwitch,
+  isPasswordError,
+}) => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+
+  const [isEmailValid, setIsEmailValid] = useState(false);
+  const [isPasswordValid, setIsPasswordValid] = useState(false);
+
+  const [isButtonEnabled, setIsButtonEnabled] = useState(false);
+
+  const handleEmailChange = (e) => {
+    const isEmailValid = e.target.validity.valid;
+    setIsEmailValid(isEmailValid);
+    setEmail(e.target.value);
+    setIsButtonEnabled(isEmailValid && isPasswordValid);
+  };
+
+  const handlePasswordChange = (e) => {
+    const isPasswordValid = e.target.validity.valid;
+    setIsPasswordValid(isPasswordValid);
+    setPassword(e.target.value);
+    setIsButtonEnabled(isEmailValid && isPasswordValid);
```

Comment:
If it’s interesting for you here is a custom `validation hook`.

You should put it in a separate file  `useFormAndValidation.js` in folder `hooks` and then import it to the component that needs validation:

```js

import {useState, useCallback} from 'react';

export function useFormAndValidation() {
  const [ values, setValues ] = useState({});
  const [ errors, setErrors ] = useState({});
  const [ isValid, setIsValid ] = useState(true);

  const handleChange = (e) => {
    const {name, value} = e.target
    setValues({...values, [name]: value });
    setErrors({...errors, [name]: e.target.validationMessage});
    setIsValid(e.target.closest('form').checkValidity());
  };



  const resetForm = useCallback((newValues = {}, newErrors = {}, newIsValid = false) => {
    setValues(newValues);
    setErrors(newErrors);
    setIsValid(newIsValid);
  }, [setValues, setErrors, setIsValid]);

  return { values, handleChange, errors, isValid, resetForm, setValues, setIsValid };
}


```

And only one line of code enables validation:

```js

  const {values, handleChange, errors, isValid, setValues, resetForm} = useFormAndValidation()

```

So, you **don’t need** to make functions for handling inputs, storing values, resetting the form. Everything is already done.